cognite_toolkit/_cdf_tk/utils/dtype_conversion.py

```diff
@@ -0,0 +1,266 @@
+import ctypes
+import json
+from abc import ABC, abstractmethod
+from collections.abc import Mapping
+from typing import ClassVar, cast
+
+from cognite.client.data_classes.data_modeling.data_types import Enum, ListablePropertyType
+from cognite.client.data_classes.data_modeling.instances import PropertyValueWrite
+from cognite.client.data_classes.data_modeling.views import PropertyType
+from cognite.client.utils import ms_to_datetime
+from dateutil import parser
+
+from cognite_toolkit._cdf_tk.exceptions import ToolkitNotSupported
+
+from .collection import humanize_collection
+
+INT32_MIN = -2_147_483_648
+INT32_MAX = 2_147_483_647
+INT64_MIN = -9_223_372_036_854_775_808
+INT64_MAX = 9_223_372_036_854_775_807
+
+
+def convert_to_primary_property(
+    value: str | int | float | bool | dict | list | None, type_: PropertyType, nullable: bool
+) -> PropertyValueWrite:
+    """Convert a string value to the appropriate type based on the provided property type.
+
+    Args:
+        value (str | int | float | bool): The value to convert.
+        type_ (PropertyType): The type of the property to convert to.
+        nullable (bool): Whether the property can be null.
+
+    Returns:
+        PropertyValueWrite: The converted value as a PropertyValue, or None if is_nullable is True and the value is empty.
+    """
+    dtype = type_._type
```

Comment:
nit: would be nice to not rely on a private property. At a glance it looks like there's not a good alternative though.

Reply:
Agreed, unfortunately I do not see any other way to do it.